core/include/xfa.h

```diff
@@ -158,14 +158,20 @@ _Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
  *
  * Pointers will end up sorted by prio.
  *
+ * @note This implementation uses the __typeof__() C extension.
+ *       It is available both in GCC and LLCM, and both don't complain with
+ *       plain "-std=c11". If deemed necessary, use of __typeof__ can be removed
+ *       and the type can be added as parameter, at the cost of much less
+ *       convenience.
+ *
  * @param[in]   xfa_name    name of the xfa
  * @param[in]   prio        priority within the xfa
  * @param[in]   name        symbol name
  * @param[in]   entry       pointer variable to add to xfa
  */
```

Comment:
See https://github.com/RIOT-OS/RIOT/pull/15795 to get proper fix proposals here ;-)